### PR TITLE
Add dynamically set url

### PR DESCRIPTION
**Pull request by joshxg**
Can now set the Grafana url from the UI

### CODE REVIEW

File: backend/pkg/api/dashboard.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func NewDashboardResponse(response *http.Response) (*DashboardResponse, error) {
 }
 
 func NewDashboard(authConfig *auth.AuthConfig, uuid string, from string, to string, datasourceID int) (*Dashboard, error) {
-	url := "http://" + authConfig.AuthString() + "localhost:3000/api/dashboards/uid/" + uuid
+	url := "http://" + authConfig.AuthString() + authConfig.URL + "/api/dashboards/uid/" + uuid
```
Comment (wlthomson) [inline backend/pkg/api/dashboard.go:161]: Would this be better done using an `Authorization` header? I think URL encoding auth strings has been deprecated in the HTTP standard (not 100% about this, but I recall reading it).

File: backend/pkg/dbstore/settings.go [hunk 2]
```diff
@@ -116,22 +118,22 @@ func (datasource *SQLiteDatasource) GetSettings() (*Settings, error) {
...
 		rows.Next()
-		err = rows.Scan(&id, &grafanaUsername, &grafanaPassword, &email, &emailPassword, &datasourceID, &emailHost, &emailPort)
+		err = rows.Scan(&id, &grafanaUsername, &grafanaPassword, &email, &emailPassword, &datasourceID, &emailHost, &emailPort, &grafanaURL)
 		if err != nil {
 			log.DefaultLogger.Error("GetSettings: rows.Scan(): ", err.Error())
 			return nil, err
 		}
 
-		return &Settings{GrafanaUsername: grafanaUsername, GrafanaPassword: grafanaPassword, Email: email, EmailPassword: emailPassword, DatasourceID: datasourceID, EmailPort: emailPort, EmailHost: emailHost}, nil
+		return &Settings{GrafanaUsername: grafanaUsername, GrafanaPassword: grafanaPassword, Email: email, EmailPassword: emailPassword, DatasourceID: datasourceID, EmailPort: emailPort, EmailHost: emailHost, GrafanaURL: grafanaURL}, nil
```
Comment (wlthomson) [inline backend/pkg/dbstore/settings.go:135]: Could add a `Settings` constructor?

File: frontend/src/components/PanelList.tsx [hunk 3]
```diff
@@ -41,14 +41,15 @@ const marginForCheckbox = css`
 
 type Props = {
   schedule: Schedule;
+  datasourceID: number;
 };
 
 // TODO: Create PanelItem component and extract logic there.
-export const PanelList: FC<Props> = ({ schedule }) => {
+export const PanelList: FC<Props> = ({ schedule, datasourceID }) => {
```
Comment (wlthomson) [inline frontend/src/components/PanelList.tsx:48]: Just a style I've picked up, I like to do:

```
export type PanelList = React.FunctionalComponent<PanelListProps>;

export const PanelList: PanelList = ...
```

Benefit of this is that you can export the props used by a component, which is useful in typechecking variables in components higher up the tree. YMMV 😄 .

**Review by wlthomson — APPROVED** [review]
lgtm

File: backend/pkg/api/dashboard.go (hunk 1 above)
Comment (joshxg) [reply backend/pkg/api/dashboard.go:161]: It's probably the worst way to do it and any other way would be better, yup

File: backend/pkg/dbstore/settings.go (hunk 2 above)
Comment (joshxg) [reply backend/pkg/dbstore/settings.go:135]: yeah will do

File: frontend/src/components/PanelList.tsx (hunk 3 above)
Comment (joshxg) [reply frontend/src/components/PanelList.tsx:48]: aren't props checked anyway when passing them?

Comment (wlthomson) [reply frontend/src/components/PanelList.tsx:48]: Yeah, but I guess it's just for neatness in the calling code if maybe you want to "construct" the props in the higher up component, e.g. define an object for the props, somethinglike:

```
panelProps: PanelListProps = { ... };
return <PanelList {,,,panelProps} />
```

or whatever!

**Pull request merged**